app/models/hardware.rb

```diff
@@ -44,6 +43,11 @@ def mac_addresses
   def ram_size_in_bytes
     memory_mb.to_i * 1.megabyte
   end
+  # resulting sql: "(CAST("hardwares"."memory_mb" AS bigint) * 1048576)"
+  virtual_attribute :ram_size_in_bytes, :integer, :arel => (lambda do |t|
+    t.grouping(Arel::Nodes::Multiplication.new(
+      Arel::Nodes::NamedFunction.new("CAST", [t[:memory_mb].as("bigint")]), 1.megabyte))
+  end)
```

Comment:
@yrudman I would say write a test that creates a hardware record with the same value as what you have in the screenshot you just share, and then try calling the `ram_size_in_bytes` method on that record before and after this change.  Should break in master and then be fixed with your change.

So, you should just be able to use the console you have open already, and run `ActiveRecord::Base.connection.select_values("SELECT MAX(hardwares.memory_mb) FROM hardwares")` to get your out of range value to plug into a spec.  The spec should be something simple like this:

```ruby
describe "#ram_size_in_bytes" do
  it "should not cause out of range errors" do
    FactoryGirl.create(:hardware, :memory_mb => <<VALUE_YOU_GOT_FROM_THE_CONSOLE_ABOVE>>)
    expect{ Hardware.select(:ram_size_in_bytes).all }.to_not raise_error(ActiveRecord::StatementInvalid)
  end
end
```